libs/libcommon/include/common/logger_useful.h

```diff
@@ -60,61 +50,58 @@ std::string toCheckedFmtStr(const S & format, const Ignored &, Args &&... args)
-        }                                                                           \
+#define LOG_INTERNAL(logger, PRIORITY, message)                             \
+    do                                                                      \
+    {                                                                       \
+        Poco::Message poco_message(                                         \
+            /*source*/ (logger)->name(),                                    \
+            /*text*/ (message),                                             \
+            /*prio*/ (PRIORITY),                                            \
+            /*file*/ &__FILE__[LogFmtDetails::getFileNameOffset(__FILE__)], \
+            /*line*/ __LINE__);                                             \
+        (logger)->log(poco_message);                                        \
     } while (false)
 
-#define LOG_FMT_TRACE(logger, ...) LOG_FMT_IMPL(logger, Poco::Message::PRIO_TRACE, __VA_ARGS__)
-#define LOG_FMT_DEBUG(logger, ...) LOG_FMT_IMPL(logger, Poco::Message::PRIO_DEBUG, __VA_ARGS__)
-#define LOG_FMT_INFO(logger, ...) LOG_FMT_IMPL(logger, Poco::Message::PRIO_INFORMATION, __VA_ARGS__)
-#define LOG_FMT_WARNING(logger, ...) LOG_FMT_IMPL(logger, Poco::Message::PRIO_WARNING, __VA_ARGS__)
-#define LOG_FMT_ERROR(logger, ...) LOG_FMT_IMPL(logger, Poco::Message::PRIO_ERROR, __VA_ARGS__)
-#define LOG_FMT_FATAL(logger, ...) LOG_FMT_IMPL(logger, Poco::Message::PRIO_FATAL, __VA_ARGS__)
+
+#define LOG_IMPL_0(logger, PRIORITY, message)        \
+    do                                               \
+    {                                                \
+        if ((logger)->is((PRIORITY)))                \
+            LOG_INTERNAL(logger, PRIORITY, message); \
+    } while (false)
+
+#define LOG_IMPL_1(logger, PRIORITY, fmt_str, ...)                                            \
+    do                                                                                        \
+    {                                                                                         \
+        if ((logger)->is((PRIORITY)))                                                         \
+        {                                                                                     \
+            auto _message = LogFmtDetails::toCheckedFmtStr(FMT_STRING(fmt_str), __VA_ARGS__); \
+            LOG_INTERNAL(logger, PRIORITY, _message);                                         \
+        }                                                                                     \
+    } while (false)
+
+#define LOG_IMPL_CHOSER(...) TF_GET_29TH_ARG(__VA_ARGS__, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_1, LOG_IMPL_0)
```

Comment:
Could you tell me why have 29 args？

Reply:
Because in our repo the maximum of arguments count is 26.